File: src/main/java/tooko/td/client/TdException.java
package tooko.td.client;

import org.jetbrains.annotations.*;
import tooko.td.*;

public class TdException extends RuntimeException {

    private TdApi.Error error;

    public TdException(TdApi.Error error) {

        this.error = error;

    }

    public TdException(int code, String message) {

        this(new TdApi.Error(code, message));

    }


    public TdException(String message) {

        this(-1, message);

    }

    public TdApi.Error getError() {

        return error;

    }

    public int getCode() {

        return error.code;

    }

    @Override
    @NotNull
    public String getMessage() {

        return error.message;

    }

    @Override
    public String toString() {

        return getCode() + " : " + getMessage();

    }


}

File: src/main/java/tookox/core/client/TdException.kt
package tookox.core.client

import td.TdApi.Error

class TdException(val error: Error) : RuntimeException() {

    constructor(code: Int, message: String) : this(Error(code, message))

    constructor(message: String) : this(-1, message)

    val code: Int
        get() = error.code

    override val message: String
        get() = error.message

    override fun toString(): String {
        return "$code : $message"
    }

}